### ISSUE TITLE
Any channel pruning examples?

### ISSUE THREAD

**Issue by zeakey** [dmlc/gluon-cv#865]
Hi there, I'm trying some channel pruning ideas in mxnet.
However it's hard to get some simple tutorials or examples I can start with.

So is there some basic examples showing how to do channel pruning (freezing)
during training?

An example of channel pruning is **Learning Efficient Convolutional Networks through Network Slimming** [arXiv](arxiv.org/abs/1708.06519) [PyTorch code](https://github.com/foolwood/pytorch-slimming).
And the example of channel freezing is **Soft Filter Pruning for Accelerating Deep Convolutional Neural Networks** [arXiv](arxiv.org/abs/1808.06866) [PyTorch code](https://github.com/he-y/soft-filter-pruning).

By channel pruning I mean eliminate some of the neurons (channels) during training.
By channel freezing I mean zero-mask the data (in forward) and the gradient (in backward)  of part of the neurons during training.

I also raised a discussion here <https://discuss.gluon.ai/t/topic/13395>.

I just immigrated to MXNet from PyTorch so my main problem is that I'm not very familiar with
the API. For example I don't know how to access the data and gradient of model parameters.

____
Update-0:
I started to implement by myself. In the document of `Parameter` (<https://mxnet.incubator.apache.org/_modules/mxnet/gluon/parameter.html>), there only the
`set_data()` method. The `set_grad()` method is absent through which I can freeze some of the parameters by zeroing their gradients.
So how can I modify the parameter gradient manually?
